agent/provider/src/provider_agent.rs

```diff
@@ -75,6 +76,7 @@ pub struct ProviderAgent {
     log_handler: LoggerHandle,
     networks: Vec<NetworkName>,
     keystore_monitor: FileMonitor,
+    net_api: NetApi,
```

Comment:
I placed `net_api` directly instead of turning it into `Actor` because it is used only in one async call [here ](https://github.com/pwalski/yagna/blob/839af3cd3e2074189e468dc5cf5f71dc530cb3f8/agent/provider/src/provider_agent.rs#L541)